lib/dal/src/func/authoring/save.rs

```diff
@@ -462,13 +468,21 @@ async fn create_new_attribute_prototype(
     ctx: &DalContext,
     prototype_bag: &AttributePrototypeBag,
     func_id: FuncId,
+    key: Option<String>,
 ) -> FuncAuthoringResult<AttributePrototypeId> {
     let attribute_prototype = AttributePrototype::new(ctx, func_id).await?;
 
+    // TODO(nick): just destroy and burn nilId to the ground. We need to use the "id!" macro instead
+    // of the "pk!" macro and be done with it.
+    let component_id_cannot_be_nil_id = match prototype_bag.component_id {
+        None | Some(ComponentId::NONE) => None,
+        Some(component_id) => Some(component_id),
+    };
```

Comment:
This will need a further look, but I am out of time. Next time I'm back? Noah's ark to the `nilId`.